src/js/modules/utilities/Logging.ts

```diff
@@ -9,40 +9,99 @@
  */
 
 import * as winston from "winston";
+import { Container, Logger, LoggerOptions } from "winston";
 
-export function newLogger(
-  module_name: string,
-  logLevel = "info",
-  simplify = false
-) {
-  const rpWasmFormatter = () => {
-    const logLineFormatter = winston.format.printf(
-      ({ level, message, label, timestamp }) => {
-        return `${timestamp} [${label}] ${level}: ${message}`;
-      }
-    );
-    const simpleLogLineFormatter = winston.format.printf(
-      ({ level, message, label }) => {
-        return `[${label}] ${level}: ${message}`;
-      }
-    );
+const maxLogFileSize = 5000000;
+
+const rpWasmFormatter = (moduleName: string, simplify: boolean) => {
+  const logLineFormatter = winston.format.printf(
+    ({ level, message, label, timestamp }) => {
+      return `${timestamp} [${label}] ${level}: ${message}`;
+    }
+  );
+  const simpleLogLineFormatter = winston.format.printf(
+    ({ level, message, label }) => {
+      return `[${label}] ${level}: ${message}`;
+    }
+  );
+
+  return (simplify ? [] : [winston.format.timestamp()]).concat([
+    winston.format.splat(),
+    winston.format.label({ label: moduleName }),
+    simplify ? simpleLogLineFormatter : logLineFormatter,
+  ]);
+};
+
+export class LogService {
+  private static instance: LogService;
+  private logFilePath: string;
+  private winstonContext: Container;
+
+  constructor() {
+    this.winstonContext = new winston.Container();
+  }
+
+  static getInstance(): LogService {
+    if (this.instance == undefined) {
+      this.instance = new LogService();
+      return this.instance;
+    }
+    return this.instance;
+  }
 
-    return (simplify ? [] : [winston.format.timestamp()]).concat([
-      winston.format.splat(),
-      winston.format.label({ label: module_name }),
-      simplify ? simpleLogLineFormatter : logLineFormatter,
-    ]);
-  };
-
-  const logger = winston.createLogger({
-    level: logLevel,
-    format: winston.format.combine(...rpWasmFormatter()),
-    defaultMeta: { service: module_name },
-    transports: [
-      new winston.transports.Console({
-        level: logLevel,
-      }),
-    ],
-  });
-  return logger;
+  setPath(path: string): void {
```

Comment:
whats the expected behavior of setPath? Should it modify the log path of all loggers or just newly added ones?

Reply:
Nope, this property just be set one time, in order to guarantee all logs have the same path file.

Reply:
Then it should be a constructor parameter

Reply:
No, because it is a singleton implementation, I can start this with that value

Reply:
I set this value after reading yaml config

Reply:
Relevant: https://github.com/vectorizedio/redpanda/issues/353